src/isDeepEqual.ts

```diff
@@ -174,3 +178,33 @@ function isDeepEqualMaps(
 
   return true;
 }
+
+function isDeepEqualSets(
+  data: ReadonlySet<unknown>,
+  other: ReadonlySet<unknown>,
+): boolean {
+  if (data.size !== other.size) {
+    return false;
+  }
+
+  // TODO: Once we bump our typescript target we can iterate over the map keys and values directly.
+  const dataArr = Array.from(data.values());
+  const otherArr = Array.from(other.values());
+
+  for (const dataItem of dataArr) {
+    let isFound = false;
+
+    for (let i = 0; i < otherArr.length; i++) {
+      if (isDeepEqualImplementation(dataItem, otherArr[i])) {
+        isFound = true;
+        otherArr.splice(i, i);
```

Comment:
This is incorrect, the second operand for slice is the number of items to remove, it should be 
```suggestion
        otherArr.splice(i, 1);
```

This also means the test plan doesn't cover cases where this would have broken. Please also add a test that tests sets with more than 2 items to make sure the code works as intended.